rfcs/2022-02-11-responsive-and-adaptive-layout-strategy.md

```diff
@@ -0,0 +1,391 @@
+
+1. Larger Screen Size - Search Input with Autosuggestion
+2. Smaller Screen Size - Search Field when clicked opens up a full screen modal which basically gives an optimised search experience on smaller screen size.
+
+<img alt="Responsive Modal" src="./images/responsive-adaptive-layout/adaptive-search-field.png">
+
+# Motivation
+
+### Why are we doing this?
+
+As of today, we don't have any strategy or approach in place for designing layouts for different screen sizes. Ensuring that elements fit within a page is not enough. There are different experiences that works for different screen sizes.
+
+Just going Responsive(Fluid) Design blindly is not correct. Imagine for example, you have different experiences(styles, javascript) for mobile vs desktop but if you just go blindly with responsive we might unnecessary download assets for all the screen sizes our application would run on. For example it doesn't makes sense to implement [this Search Field](#adaptive-search-field) as a responsive layout since it'll result in unnecessary download of the assets(modal for mobile) and hence hampering the experience and performance for the consumers.
+
+Similarly, just going adaptive blindly is not correct too. We might unnecessary be creating redundant variant of the same component. For example, imagine that we are creating 2 variants of a modal - 1 for desktop(which has a width of `600px`) and 1 for mobile(which has `100%` width). Now we could have just made it responsive by setting the `maxWidth` property of the modal so it could have been `600px` in width on desktop and `100%` in width on mobile. For eg, it doesn't makes sense to create multiple variant of [this Modal](#responsive-modal) while we can achieve this through responsive layout by setting `maxWidth` for Modal's Dialog component.
+
+But because we don't have these strategies in place we don't know how to approach building layouts and components for different screen sizes.
+
+### What use cases does it support?
+
+- Handling responsiveness internally in the components based on different screen sizes.
+- Layout components that will help you define different layouts based on screen sizes.
+- Rendering different variants of a component based on screen sizes, optimising for performance without bloating the bundlesize.
+
+### What is the expected outcome?
+
+We'll basically define the approach and strategy we'll be taking in Blade to build components that will address the concerns of rendering based on different screen sizes.
+
+# Detailed Design
+
+## Responsive Design
+
+- Responsive is fluid and adapts to the size of the screen no matter what the target device is.
+- Responsive uses [CSS media queries](https://developer.mozilla.org/en-US/docs/Web/CSS/Media_Queries/Using_media_queries) to change styles based on the target device such as display type, width, height, etc., and only one of these is necessary for the site to adapt to different screens.
```

Comment:
We can alt-code these in the statement:
display `type`, `width`, `height`, etc.